### PR TITLE
Pass `EndOfRouteFeedback` to the app

### PR DESCRIPTION
**Pull request by azarovalex**
### Description
Added a new method to `NavigationViewControllerDelegate` which is called to notify that the user submitted the end of route feedback.

### CODE REVIEW

File: Sources/MapboxNavigation/NavigationViewController.swift [hunk 1]
```diff
@@ -455,7 +455,7 @@ open class NavigationViewController: UIViewController, NavigationStatusPresenter
         return true
     }
     
-    fileprivate func handleCancelAction() {
+    fileprivate func handleCancelAction(with feedback: EndOfRouteFeedback? = nil) {
```
Comment (bamx23) [inline Sources/MapboxNavigation/NavigationViewController.swift:458]: Where is this `feedback` used?

File: Sources/MapboxNavigation/ArrivalController.swift [hunk 2]
```diff
@@ -4,10 +4,14 @@ import MapboxCoreNavigation
 import MapboxMobileEvents
 import MapboxMaps
 
+public protocol ArrivalControllerProtocol {
```
Comment (bamx23) [inline Sources/MapboxNavigation/ArrivalController.swift:7]: Is it used?

File: Example/ViewController.swift [hunk 3]
```diff
@@ -621,7 +621,6 @@ class ViewController: UIViewController {
 // MARK: - NavigationMapViewDelegate methods
 
 extension ViewController: NavigationMapViewDelegate {
-
```
Comment (bamx23) [inline Example/ViewController.swift:624]: Unnecessary change

File: Sources/MapboxNavigation/ArrivalController.swift (hunk 2 above)
Comment (azarovalex) [reply Sources/MapboxNavigation/ArrivalController.swift:7]: No, accidentally committed, thx

File: Sources/MapboxNavigation/NavigationViewController.swift (hunk 1 above)
Comment (azarovalex) [reply Sources/MapboxNavigation/NavigationViewController.swift:458]: Not used anymore, thx, will fix

**Review by bamx23 — APPROVED** [review]
